content/docs/ethernet-v0.5/fpga.md

```diff
@@ -0,0 +1,283 @@
+    
+Convert the bitstream to quad-spi memory format
+
+    vivado -mode batch -source $TOP/fpga/common/script/cfgmem.tcl -tclargs "xc7a100t_0" nexys4ddr.bit
+
+Burn the quad-spi memory (Ensure the MODE switch is set to QSPI)
+
+    vivado -mode batch -source $TOP/fpga/common/script/program_cfgmem.tcl -tclargs "xc7a100t_0" nexys4ddr.bit.mcs
+
+## Access the boot-loader and connect to the UART console.
+
+First check the available serial ports:
+
+    ls -l /dev/*USB*
+
+which should result in a response similar to:
+
+    crw-rw---- 1 root dialout 188, 1 Apr 14 16:32 /dev/ttyUSB1
+
+In this case the device is user and group only access, so you need to be a member of the group dialout to use it.
+
+    microcom -p /dev/ttyUSB1
+
+This needs a separate terminal window as it takes over the screen. Alternatively a separate VGA display and USB keyboard may be used. You should see a display similar to the below a short delay after the PROG button is pushed.
+
+    Selftest iteration 1
+    Selftest matches=4/4, delay = 5
+    Selftest iteration 2
+    Selftest matches=8/8, delay = 5
+    Selftest iteration 3
+    Selftest matches=16/16, delay = 7
+    Selftest iteration 4
+    Selftest matches=32/32, delay = 13
+    Selftest iteration 5
+    Selftest matches=64/64, delay = 26
+    Selftest iteration 6
+    Selftest matches=128/128, delay = 52
+    Selftest iteration 7
+    Selftest matches=256/256, delay = 103
+    Selftest iteration 8
+    Selftest matches=375/375, delay = 151
+    Hello LowRISC! Thu Dec 21 18:02:20 2017
+    MAC = eee1:e2e3e4e5
+    MAC address = ee:e1:e2:e3:e4:e5.
+    IP Address:  192.168.0.51
+    Subnet Mask: 255.255.255.0
+    Enabling interrupts
+
+## Ethernet server preparation
+
+We need to make some changes on the server to enable this option. First of all create a suitable NFSroot mount point:
+
+    sudo mkdir /mnt/poky-dev
+    sudo mount -t ext2 -o loop rootfs.ext2 /mnt/poky-dev
+    sudo vi /etc/xinetd.d/time
+    #Change the first occurence of "disable=yes" to "disable=no". This enables the time service.
+    sudo /etc/init.d/xinetd restart
+    sudo /etc/init.d/nfs-kernel-server stop
+    sudo vi /etc/exports
+    #Add the following to the end of the file. "/mnt/poky-dev 192.168.0.51(rw,sync,no_root_squash)"
+    #Mount the previously downloaded rootfs onto the designated mount point (stored in initial_0001 in riscv-tools)
+    sudo mount -t ext2 -o loop rootfs.ext2 /mnt/poky-dev
+    #Restart the NFS server
+    sudo /etc/init.d/nfs-kernel-server start
+
+The dangerous option no_root_squash which allows the NFS client to run as root on the server, is mitigated by the confinement of the mount to the area of the loopback which we have set up. If access to host files is desired this may be done by copying on the host, or with a second mount point without no_root_squash inside the server's filing system
+
+## Ethernet booting
+
```

Comment:
Don't you need to configure the ethernet interface you're using to connect to the FPGA board appropriately?